parser/agent.go

```diff
@@ -87,18 +87,20 @@ func findErrorVariable(stmt *dst.AssignStmt, pkg *decorator.Package) dst.Expr {
 //////////////////////////////////////////////
 
 // InstrumentMain looks for the main method of a program, and uses this as an instrumentation initialization and injection point
-// TODO: Can this be refactored to be part of the Trace Function algorithm?
 func InstrumentMain(manager *InstrumentationManager, c *dstutil.Cursor) {
 	mainFunctionNode := c.Node()
 	if decl, ok := mainFunctionNode.(*dst.FuncDecl); ok {
-		// only inject go agent into the main.main function
+		// Check functions return signatures for newrelic.Application and if it exists, load it into manager.setupFunc
+		// We don't want to propagate tracing into the setup function so later on in our trace function we will ignore it
+		CheckForExistingApplicationInFunctions(manager, decl, c)
```

Comment:
Should we make this check after the `if decl.Name.Name == "main"` check below to avoid doing unnecessary work, or is this `main`-agnostic?